commit 832855354b5863d141ecfc563e78063978c6f29f
Author: Dan Carpenter <error27@gmail.com>
Date:   Fri Nov 12 08:11:13 2010 +0300

    Staging: rtl8712: signedness bug in init
    
    PollingCnt is 20 and that means we loop 20 times and then run the
    timeout code.  After the end of the loop PollingCnt should be -1 but
    because it's an unsigned char, it's actually 255 and the timeout
    code never runs.
    
    Signed-off-by: Dan Carpenter <error27@gmail.com>
    Acked-by: Larry Finger <Larry.Finger@lwfinger.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/rtl8712/usb_halinit.c
+++ b/drivers/staging/rtl8712/usb_halinit.c
@@ -37,7 +37,7 @@ u8 r8712_usb_hal_bus_init(struct _adapte
 {
 	u8 val8 = 0;
 	u8 ret = _SUCCESS;
-	u8 PollingCnt = 20;
+	int PollingCnt = 20;
 	struct registry_priv *pregistrypriv = &padapter->registrypriv;
 
 	if (pregistrypriv->chip_version == RTL8712_FPGA) {
